backend/src/routes/api/storage/storageUtils.ts

```diff
@@ -0,0 +1,96 @@
+import { Client as MinioClient } from 'minio';
+import { DSPipelineKind, KubeFastifyInstance } from '../../../types';
+import { Transform, PassThrough } from 'stream';
+
+/**
+ * Create minio client with aws instance profile credentials if needed.
+ * @param config minio client options where `accessKey` and `secretKey` are optional.
+ */
+export async function createMinioClient(
+  fastify: KubeFastifyInstance,
+  namespace: string,
+): Promise<MinioClient> {
+  try {
+    const dspaResponse = await fastify.kube.customObjectsApi
```

Comment:
This is problematic -- this is done with the service account. We need to do this with the user's token.

Otherwise you could provide a namespace you do not have access to and get back a response.

See the `utils/proxy.ts` for an example.